tests/compiler/std/string.ts

```diff
@@ -80,27 +81,206 @@ assert("ab c".trim() == "ab c");
+assert(parseFloat("\u2029\u2029-1.1") == -1.1);
+assert(parseFloat("00000") == 0.0);
+assert(parseFloat("0000a") == 0.0);
+assert(parseFloat("00001") == 1.0);
+assert(parseFloat("00.00") == 0.0);
+assert(parseFloat("00.0a") == 0.0);
+assert(parseFloat("1e1e") == 10.0);
+assert(parseFloat("1e+0001") == 10.0);
+assert(parseFloat("0e+100") == 0.0);
+assert(parseFloat("1.-1.") == 1.0);
+assert(parseFloat("1e-1.2") == 0.1);
+assert(parseFloat("1ex") == 1.0);
+assert(parseFloat("1e1x") == 10);
+assert(parseFloat("1e-x") == 1.0);
+assert(parseFloat("1e-1x") == 0.1);
+assert(parseFloat("0.1e-1x") == 0.01);
+assert(parseFloat("00.") == 0.0);
+assert(parseFloat(".00") == 0.0);
+assert(parseFloat(".0.") == 0.0);
+assert(parseFloat(".1.") == 0.1);
+assert(parseFloat("0..") == 0.0);
+assert(parseFloat("0.a") == 0.0);
+assert(parseFloat("1..1") == 1.0);
+assert(parseFloat("0.1.1") == 0.1);
+assert(parseFloat("0. 1") == +0.0);
+assert(parseFloat("+0.0") == +0.0);
+assert(parseFloat("-0.0") == -0.0);
+assert(parseFloat("+0") == +0);
+assert(parseFloat("-0") == -0);
+assert(isNaN(parseFloat("+")));
+assert(isNaN(parseFloat("-")));
+assert(isNaN(parseFloat("--0")));
+assert(isNaN(parseFloat("++0")));
+assert(isNaN(parseFloat(".a")));
+assert(isNaN(parseFloat("..0")));
+assert(isNaN(parseFloat(".")));
+assert(isNaN(parseFloat("..")));
+assert(isNaN(parseFloat("NaN")));
+assert(isNaN(parseFloat("\u000B")));
+assert(isNaN(parseFloat("\u180E")));
+assert(isNaN(parseFloat("\u180E1.1")));
+assert(isNaN(parseFloat("\u180E\u180E1.1")));
+assert(isNaN(parseFloat("\u000C")));
+assert(isNaN(parseFloat("true")));
+assert(isNaN(parseFloat("false")));
+assert(parseFloat("2.220446049250313e-16") == f64.EPSILON);
+assert(parseFloat("1.7976931348623157e+308") == f64.MAX_VALUE);
+assert(parseFloat("5e-324") == f64.MIN_VALUE);
+assert(parseFloat("0.000001e+314") == 1e+308);
+assert(parseFloat("0.00000000000000000000000000000000000000000000000000000001e+56") == 1.0);
+assert(parseFloat("+1E-325") == 0.0);
+assert(parseFloat("+1E+309") == Infinity);
+assert(parseFloat("-1E-325") == 0.0);
+assert(parseFloat("-1E+309") == -Infinity);
+assert(parseFloat("1e-1000000") == 0.0);
+assert(parseFloat("1e+1000000") == Infinity);
+assert(parseFloat("1.e360") == Infinity);
+assert(parseFloat(" Infinity") == Infinity);
+assert(parseFloat("+Infinity") == Infinity);
+assert(parseFloat("-Infinity") == -Infinity);
+assert(parseFloat("Infinityx") == Infinity);
+assert(parseFloat("Infinity+1") == Infinity);
+assert(isNaN(parseFloat("Infi")));
+assert(isNaN(parseFloat("+Infinit")));
+assert(isNaN(parseFloat("infinity")));
+
+// precision tests
+// 2^-1074 * 0.5 - eps
+assert(parseFloat(".2470328229206232720882843964341106861825299013071623822127928412503377536351043e-323") == 0);
+
+// 2^-1074 * 0.5 + eps
+// const Ox1p_1074 = reinterpret<f64>(0x1); // 0x1p-1074
+// assert(parseFloat(".2470328229206232720882843964341106861825299013071623822127928412503377536351044e-323") == Ox1p_1074); // FAIL
```

Comment:
What's the implication of these cases failing?
* 2^-1074 * 0.5 + eps
* 2^-1074 * 1.5 - eps
* 2^1024 - 2^970

Reply:
Will add comment

Reply:
For 2^1024 - 2^970 we have already comment btw